torch/distributed/_tensor/ops/tensor_ops.py

```diff
@@ -472,3 +472,93 @@ def place(vp: Placement, ip: Placement) -> Placement:
             ],
         )
         return result
+
+
+@register_prop_rule("aten.cat.default")
+def cat_rule(op_schema: OpSchema) -> OutputSharding:
+    dim = 0  # default dim = 0
+    tensor_list_specs = cast(List[DTensorSpec], op_schema.args_schema[0])
+    if (len(op_schema.args_schema) > 1):
+        dim = cast(int, op_schema.args_schema[1])
+    # normalize arguments
+    if dim < 0:
+        dim += tensor_list_specs[0].ndim
+
+    # check concat dim
+    needs_reshard_on_cat_dim = False
+    for spec in tensor_list_specs:
+        if dim < len(spec.placements) and spec.placements[dim].is_shard():
+            needs_reshard_on_cat_dim = True
+            spec.placements = unshard_tensor_dim(spec.placements, dim=dim)
```

Comment:
please don't modify the original schema and its specs inplace, create a new schema with the expected placements.